src/PAModel/MsAppTest.cs

```diff
@@ -14,6 +16,164 @@ namespace Microsoft.PowerPlatform.Formulas.Tools
+
+            return HasNoDeltas(doc1, docClone, strict: true);
+        }
+
+
+        public static bool DiffStressTest(string pathToMsApp)
+        {
+            (CanvasDocument doc1, var errors) = CanvasDocument.LoadFromMsapp(pathToMsApp);
+            errors.ThrowOnErrors();
+
+            return HasNoDeltas(doc1, doc1);
+        }
+
+        // Verify there are no deltas (detected via smart merge) between doc1 and doc2
+        // Strict =true, also compare entropy files. 
+        private static bool HasNoDeltas(CanvasDocument doc1, CanvasDocument doc2, bool strict = false)
+        {
+            var ourDeltas = Diff.ComputeDelta(doc1, doc1);
+
+            // ThemeDelta always added
+            ourDeltas = ourDeltas.Where(x => x.GetType() != typeof(ThemeChange)).ToArray();
+
+            if (ourDeltas.Any())
+            {
+                foreach (var diff in ourDeltas)
+                {
+                    Console.WriteLine($"  {diff.GetType().Name}");
+                }
+                // Error! app shouldn't have any diffs with itself.
+                return false;
+            }
+
+
+            // Save and verify checksums.
+            using (var temp1 = new TempFile())
+            using (var temp2 = new TempFile())
+            {
+                doc1.SaveToMsApp(temp1.FullPath);
+                doc2.SaveToMsApp(temp2.FullPath);
+
+                // bool same = Compare(doc1, doc2, Console.Out);
+                bool same;
+                if (strict)
+                {
+                    same = Compare(temp1.FullPath, temp2.FullPath, Console.Out);
+                }
+                else
+                {
+                    var doc1NoEntropy = RemoveEntropy(temp1.FullPath);
+                    var doc2NoEntropy = RemoveEntropy(temp2.FullPath);
+
+                    same = Compare(doc1NoEntropy, doc2NoEntropy, Console.Out);
+                }
+
+                if (!same)
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        // Unpack, delete the entropy dirs, repack. 
+        public static CanvasDocument RemoveEntropy(string pathToMsApp)
+        {
+            using (var temp1 = new TempDir())
+            {
+                (CanvasDocument doc1, var errors) = CanvasDocument.LoadFromMsapp(pathToMsApp);
+                errors.ThrowOnErrors();
+
+                doc1.SaveToSources(temp1.Dir);
+
+                var entropyDir = Path.Combine(temp1.Dir, "Entropy");
+                if (!Directory.Exists(entropyDir))
+                {
+                    throw new Exception($"Missing entropy dir: " + entropyDir);
+                }
+
+                Directory.Delete(entropyDir, recursive: true);
+
+                (var doc2, var errors2) = CanvasDocument.LoadFromSources(temp1.Dir);
+                errors.ThrowOnErrors();
+
+
+#if false
+                // $$$ We get different checksums when deleting in-memory vs. files.
```

Comment:
The directory also contains checksum.json and the appchecker file, in addition to what's covered by the in memory _entropy member.